File: LeetCode/Python/Medium/1428-Leftmost-Column-with-at-Least-a-One/solution.py
class Solution:
    def leftMostColumnWithOne(self, binaryMatrix: 'BinaryMatrix') -> int:
        result=-1

        dimensions=binaryMatrix.dimensions()

        rows=dimensions[0]

        columns=dimensions[1]

        for row in range(0, rows):
            left_most_column_in_row=self.search_row(binaryMatrix, row, columns)

            if left_most_column_in_row!=-1:
                if result==-1:
                    result=left_most_column_in_row
                else:
                    result=min(result, left_most_column_in_row)

        return result
    
    def search_row(self, binaryMatrix: 'BinaryMatrix', row: int, columns: int) -> int:
        result=-1
        
        low=0

        high=columns - 1

        while low <= high:
            middle=(low + (high - low)//2)

            digit=binaryMatrix.get(row, middle)

            if digit==1:
                high=middle - 1
                
                result=middle
            else:
                low=middle + 1

        return result

File: LeetCode/Rust/Medium/1428-Leftmost-Column-with-at-Least-a-One/solution.rs
impl Solution
{
    pub fn left_most_column_with_one(binaryMatrix: &BinaryMatrix) -> i32
    {
        let mut result: i32=-1;

        let dimensions: Vec<i32>=binaryMatrix.dimensions();

        let rows: usize=dimensions[0] as usize;

        let columns: usize=dimensions[1] as usize;

        for row in 0..rows
        {   
            let left_most_column_in_row: i32=Self::search_row(&binaryMatrix, columns, row);

            if left_most_column_in_row!=-1
            {
                if result==-1
                {
                    result=left_most_column_in_row;
                }
                else
                {
                    result=i32::min(result, left_most_column_in_row);
                }
            }
        }

        result
    }

    fn search_row(binaryMatrix: &BinaryMatrix, columns: usize, row: usize) -> i32
    {
        let mut left_most_column: i32=-1;

        let mut low: i32=0;

        let mut high: i32=(columns - 1) as i32;

        while low <= high
        {
            let middle: i32=(low + (high - low)/2);

            let digit: i32=binaryMatrix.get(row as i32, middle as i32);

            if digit==1
            {
                left_most_column=middle;

                high=middle - 1;
            }
            else
            {
                low=middle + 1;
            }
        }

        left_most_column
    }
}